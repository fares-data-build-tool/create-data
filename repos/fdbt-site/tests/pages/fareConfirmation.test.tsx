import * as React from 'react';
import { shallow } from 'enzyme';
import FareConfirmation, { buildFareConfirmationElements } from '../../src/pages/fareConfirmation';

describe('pages', () => {
    describe('fareConfirmation', () => {
        it('should render correctly for non school tickets', () => {
            const tree = shallow(
                <FareConfirmation
                    fareType="single"
                    passengerType={{
                        passengerType: 'adult',
                        ageRange: 'yes',
                        ageRangeMin: '18',
                        ageRangeMax: '100',
                        proof: 'yes',
                        proofDocuments: ['membership card'],
                    }}
                    groupPassengerInfo={[]}
                    schoolFareType=""
                    termTime=""
                    fullTimeRestrictions={[
                        {
                            day: 'thursday',
                            startTime: '0900',
                            endTime: '1600',
                        },
                        {
                            day: 'friday',
                            startTime: '',
                            endTime: '1600',
                        },
                        {
                            day: 'bank holiday',
                            startTime: '',
                            endTime: '',
                        },
                    ]}
                    csrfToken=""
                />,
            );
            expect(tree).toMatchSnapshot();
        });

        it('should render correctly for school tickets', () => {
            const tree = shallow(
                <FareConfirmation
                    fareType="schoolService"
                    passengerType={{
                        passengerType: 'schoolPupil',
                        ageRange: 'yes',
                        ageRangeMax: '18',
                        proof: 'yes',
                        proofDocuments: ['Student Card'],
                    }}
                    groupPassengerInfo={[]}
                    schoolFareType="single"
                    termTime="true"
                    fullTimeRestrictions={[]}
                    csrfToken=""
                />,
            );
            expect(tree).toMatchSnapshot();
        });

        describe('buildFareConfirmationElements', () => {
            it('should create confirmation elements for non school tickets', () => {
                const result = buildFareConfirmationElements(
                    'return',
                    {
                        passengerType: 'adult',
                        ageRange: 'yes',
                        ageRangeMin: '18',
                        ageRangeMax: '100',
                        proof: 'yes',
                        proofDocuments: ['membership card'],
                    },
                    [],
                    '',
                    '',
                    [
                        {
                            day: 'wednesday',
                            startTime: '0900',
                            endTime: '1600',
                        },
                        {
                            day: 'thursday',
                            startTime: '',
                            endTime: '1600',
                        },
                        {
                            day: 'friday',
                            startTime: '',
                            endTime: '',
                        },
                    ],
                );
                expect(result).toStrictEqual([
                    { content: 'Return', href: 'fareType', name: 'Fare type' },
                    { content: 'Adult', href: 'passengerType', name: 'Passenger type' },
                    {
                        content: 'Minimum age: 18 Maximum age: 100',
                        href: 'definePassengerType',
                        name: 'Passenger information - age range',
                    },
                    {
                        content: 'Membership card',
                        href: 'definePassengerType',
                        name: 'Passenger information - proof documents',
                    },
                    {
                        content: 'Start time: 0900 End time: 1600',
                        href: 'defineTimeRestrictions',
                        name: 'Time restrictions - Wednesday',
                    },
                    {
                        content: 'Start time: N/A End time: 1600',
                        href: 'defineTimeRestrictions',
                        name: 'Time restrictions - Thursday',
                    },
                    {
                        content: 'Start time: N/A End time: N/A',
                        href: 'defineTimeRestrictions',
                        name: 'Time restrictions - Friday',
                    },
                ]);
            });

            it('should create confirmation elements for school tickets', () => {
                const result = buildFareConfirmationElements(
                    'schoolService',
                    {
                        passengerType: 'schoolPupil',
                        ageRange: 'yes',
                        ageRangeMax: '18',
                        proof: 'yes',
                        proofDocuments: ['Student Card'],
                    },
                    [],
                    'single',
                    'true',
                    [],
                );
                expect(result).toStrictEqual([
                    { content: 'School service', href: 'fareType', name: 'Fare type' },
                    { content: 'School pupil', href: '', name: 'Passenger type' },
                    {
                        content: 'Minimum age: N/A Maximum age: 18',
                        href: 'definePassengerType',
                        name: 'Passenger information - age range',
                    },
                    {
                        content: 'Student card',
                        href: 'definePassengerType',
                        name: 'Passenger information - proof documents',
                    },
                    {
                        content: 'yes',
                        href: 'termTime',
                        name: 'Only valid during term times',
                    },
                    {
                        content: 'Single',
                        href: 'schoolFareType',
                        name: 'School ticket fare type',
                    },
                ]);
            });
        });
    });
});
