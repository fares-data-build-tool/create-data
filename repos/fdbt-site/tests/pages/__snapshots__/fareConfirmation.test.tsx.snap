// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages fareConfirmation should render correctly for a ticket with a premade time restriction 1`] = `
<Component
  description="Fare Confirmation page of the Create Fares Data Service"
  errors={Array []}
  title="Fare Confirmation - Create Fares Data Service"
>
  <CsrfForm
    action="/api/fareConfirmation"
    csrfToken=""
    method="post"
  >
    <h1
      className="govuk-heading-l"
    >
      Check your answers before sending your fares information
    </h1>
    <ConfirmationTable
      confirmationElements={
        Array [
          Object {
            "content": "Single",
            "href": "fareType",
            "name": "Fare type",
          },
          Object {
            "content": "Adult",
            "href": "passengerType",
            "name": "Passenger type",
          },
          Object {
            "content": "Minimum age: 18 Maximum age: 100",
            "href": "definePassengerType",
            "name": "Passenger information - age range",
          },
          Object {
            "content": "Membership card",
            "href": "definePassengerType",
            "name": "Passenger information - proof documents",
          },
          Object {
            "content": "Start time: 0900 End time: 1600",
            "href": "defineTimeRestrictions",
            "name": "Time restrictions - Thursday",
          },
          Object {
            "content": "Start time: N/A End time: 1600",
            "href": "defineTimeRestrictions",
            "name": "Time restrictions - Friday",
          },
          Object {
            "content": "Start time: N/A End time: N/A",
            "href": "defineTimeRestrictions",
            "name": "Time restrictions - Bank holiday",
          },
          Object {
            "content": "Name: Week time restrictions",
            "href": "",
            "name": "Time restriction saved for reuse",
          },
        ]
      }
      header="Fare Information"
    />
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;

exports[`pages fareConfirmation should render correctly for group tickets 1`] = `
<Component
  description="Fare Confirmation page of the Create Fares Data Service"
  errors={Array []}
  title="Fare Confirmation - Create Fares Data Service"
>
  <CsrfForm
    action="/api/fareConfirmation"
    csrfToken=""
    method="post"
  >
    <h1
      className="govuk-heading-l"
    >
      Check your answers before sending your fares information
    </h1>
    <ConfirmationTable
      confirmationElements={
        Array [
          Object {
            "content": "Single",
            "href": "fareType",
            "name": "Fare type",
          },
          Object {
            "content": "Group",
            "href": "passengerType",
            "name": "Passenger type",
          },
          Object {
            "content": "Minimum age: 16 Maximum age: 65",
            "href": "definePassengerType?groupPassengerType=adult",
            "name": "Adult passenger - age range",
          },
          Object {
            "content": "N/A",
            "href": "definePassengerType?groupPassengerType=adult",
            "name": "Adult passenger - proof documents",
          },
          Object {
            "content": "Minimum age: 0 Maximum age: 16",
            "href": "definePassengerType?groupPassengerType=child",
            "name": "Child passenger - age range",
          },
          Object {
            "content": "Identity document",
            "href": "definePassengerType?groupPassengerType=child",
            "name": "Child passenger - proof documents",
          },
          Object {
            "content": "Start time: 0900 End time: 1600",
            "href": "defineTimeRestrictions",
            "name": "Time restrictions - Thursday",
          },
          Object {
            "content": "Start time: 1400 End time: 2000",
            "href": "defineTimeRestrictions",
            "name": "Time restrictions - Thursday",
          },
          Object {
            "content": "Start time: N/A End time: 1600",
            "href": "defineTimeRestrictions",
            "name": "Time restrictions - Friday",
          },
          Object {
            "content": "Start time: N/A End time: N/A",
            "href": "defineTimeRestrictions",
            "name": "Time restrictions - Bank holiday",
          },
        ]
      }
      header="Fare Information"
    />
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;

exports[`pages fareConfirmation should render correctly for non school single tickets 1`] = `
<Component
  description="Fare Confirmation page of the Create Fares Data Service"
  errors={Array []}
  title="Fare Confirmation - Create Fares Data Service"
>
  <CsrfForm
    action="/api/fareConfirmation"
    csrfToken=""
    method="post"
  >
    <h1
      className="govuk-heading-l"
    >
      Check your answers before sending your fares information
    </h1>
    <ConfirmationTable
      confirmationElements={
        Array [
          Object {
            "content": "Single",
            "href": "fareType",
            "name": "Fare type",
          },
          Object {
            "content": "Adult",
            "href": "passengerType",
            "name": "Passenger type",
          },
          Object {
            "content": "Minimum age: 18 Maximum age: 100",
            "href": "definePassengerType",
            "name": "Passenger information - age range",
          },
          Object {
            "content": "Membership card",
            "href": "definePassengerType",
            "name": "Passenger information - proof documents",
          },
          Object {
            "content": "Start time: 0900 End time: 1600",
            "href": "defineTimeRestrictions",
            "name": "Time restrictions - Thursday",
          },
          Object {
            "content": "Start time: N/A End time: 1600",
            "href": "defineTimeRestrictions",
            "name": "Time restrictions - Friday",
          },
          Object {
            "content": "Start time: N/A End time: N/A",
            "href": "defineTimeRestrictions",
            "name": "Time restrictions - Bank holiday",
          },
        ]
      }
      header="Fare Information"
    />
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;

exports[`pages fareConfirmation should render correctly for school tickets 1`] = `
<Component
  description="Fare Confirmation page of the Create Fares Data Service"
  errors={Array []}
  title="Fare Confirmation - Create Fares Data Service"
>
  <CsrfForm
    action="/api/fareConfirmation"
    csrfToken=""
    method="post"
  >
    <h1
      className="govuk-heading-l"
    >
      Check your answers before sending your fares information
    </h1>
    <ConfirmationTable
      confirmationElements={
        Array [
          Object {
            "content": "School service",
            "href": "fareType",
            "name": "Fare type",
          },
          Object {
            "content": "School pupil",
            "href": "",
            "name": "Passenger type",
          },
          Object {
            "content": "Minimum age: N/A Maximum age: 18",
            "href": "definePassengerType",
            "name": "Passenger information - age range",
          },
          Object {
            "content": "Student card",
            "href": "definePassengerType",
            "name": "Passenger information - proof documents",
          },
          Object {
            "content": "yes",
            "href": "termTime",
            "name": "Only valid during term times",
          },
          Object {
            "content": "Single",
            "href": "schoolFareType",
            "name": "School ticket fare type",
          },
        ]
      }
      header="Fare Information"
    />
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;
